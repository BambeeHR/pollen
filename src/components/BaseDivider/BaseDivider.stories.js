import BaseDivider from './BaseDivider.vue';
import TypeOverline from '../TypeOverline/TypeOverline.vue';

export default {
  title: 'BaseDivider',
  component: BaseDivider,
  parameters: {
    componentSubtitle: 'Basic divider component.',
  },
};

export const Gallery = () => ({
  components: { BaseDivider, TypeOverline },
  template: `
  <div class="flex">
    <div class="mr-16 w-64">
      <TypeOverline tag="h1" variant="large">Horizontal</TypeOverline>
      <TypeOverline tag="h2" class="mb-4">Solid</TypeOverline>
      <BaseDivider />
      <TypeOverline tag="h2" class="mt-8 mb-4">Dashed</TypeOverline>
      <BaseDivider dashed />
      <TypeOverline tag="h2" class="mt-8 mb-4">Solid + Label</TypeOverline>
      <BaseDivider label="or" />
      <TypeOverline tag="h2" class="mt-8 mb-4">Dashed + Label</TypeOverline>
      <BaseDivider dashed label="or" />
    </div>
    <div class="w-64">
      <TypeOverline tag="h1" variant="large">Vertical</TypeOverline>
      <div class="flex items-end">
        <div class="flex-grow w-1/4">
          <TypeOverline tag="h2" class="mb-4">Solid</TypeOverline>
          <BaseDivider vertical class="h-56" />
        </div>
        <div class="flex-grow w-1/4">
          <TypeOverline tag="h2" class="mb-4">Dashed</TypeOverline>
          <BaseDivider dashed vertical class="h-56" />
        </div>
        <div class="flex-grow w-1/4">
          <TypeOverline tag="h2" class="mb-4">Solid + Label</TypeOverline>
          <BaseDivider vertical label="or" class="h-56" />
        </div>
        <div class="flex-grow w-1/4">
          <TypeOverline tag="h2" class="mb-4">Dashed + Label</TypeOverline>
          <BaseDivider dashed vertical label="or" class="h-56" />
        </div>
      </div>
    </div>
  </div>
  `,
});

/* eslint-disable no-param-reassign */
[Gallery].forEach((item) => {
  item.story = {
    parameters: {
      jest: ['BaseDivider.test.js'],
    },
  };
});
