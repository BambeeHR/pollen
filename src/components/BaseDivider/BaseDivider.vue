<template functional>
  <div
    class="base-divider"
    :class="[
      {
        'base-divider--dashed': props.dashed,
        'base-divider--solid': !props.dashed,
        'base-divider--horizontal': !props.vertical,
        'base-divider--vertical': props.vertical,
      },
      data.class,
      data.staticClass,
    ]"
    v-bind="$options.getAttributes({ data, parent })"
    role="separator"
    v-on="listeners"
  >
    <div v-if="props.label" class="base-divider__label">{{ props.label }}</div>
  </div>
</template>

<script>
import getAttributes from '../../util/getAttributes';

export default {
  getAttributes,
  props: {
    dashed: {
      type: Boolean,
      default: false,
    },
    label: {
      type: String,
      default: null,
    },
    vertical: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style scoped>
.base-divider {
  @apply flex items-center;
}

.base-divider::before,
.base-divider::after {
  @apply block flex-grow;
  content: '';
}

.base-divider--solid::before,
.base-divider--solid::after {
  @apply bg-gray-5;
}

.base-divider--horizontal.base-divider--dashed::before,
.base-divider--horizontal.base-divider--dashed::after {
  background: repeating-linear-gradient(
    90deg,
    #e0e0e0,
    #e0e0e0 2px,
    transparent 2px,
    transparent 4px
  );
}

.base-divider--vertical.base-divider--dashed::before,
.base-divider--vertical.base-divider--dashed::after {
  background: repeating-linear-gradient(
    0deg,
    #e0e0e0,
    #e0e0e0 2px,
    transparent 2px,
    transparent 4px
  );
}

.base-divider--horizontal {
  @apply h-4 w-full;
}

.base-divider--horizontal::before,
.base-divider--horizontal::after {
  height: 1px;
}

.base-divider--vertical {
  @apply flex-col h-full w-4;
}

.base-divider--vertical::before,
.base-divider--vertical::after {
  width: 1px;
}

.base-divider__label {
  @apply font-overline-small text-gray-4;
}

.base-divider--horizontal .base-divider__label {
  @apply px-2;
}

.base-divider--vertical .base-divider__label {
  @apply py-2;
}
</style>
