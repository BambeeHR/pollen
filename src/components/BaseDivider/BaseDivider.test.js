import { mount } from '@vue/test-utils';
import { Gallery } from './BaseDivider.stories';

describe('BaseDivider', () => {
  test('renders correctly', () => {
    const gallery = Gallery();
    const wrapper = mount(gallery);
    expect(wrapper.element).toMatchSnapshot();
  });
});
