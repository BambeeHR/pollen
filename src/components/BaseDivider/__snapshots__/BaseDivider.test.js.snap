// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDivider renders correctly 1`] = `
<div
  class="flex"
>
  <div
    class="mr-16 w-64"
  >
    <h1
      class="type-overline type-overline-large"
    >
      Horizontal
    </h1>
     
    <h2
      class="type-overline type-overline-small mb-4"
    >
      Solid
    </h2>
     
    <div
      class="base-divider base-divider--solid base-divider--horizontal"
      role="separator"
    >
      <!---->
    </div>
     
    <h2
      class="type-overline type-overline-small mt-8 mb-4"
    >
      Dashed
    </h2>
     
    <div
      class="base-divider base-divider--dashed base-divider--horizontal"
      role="separator"
    >
      <!---->
    </div>
     
    <h2
      class="type-overline type-overline-small mt-8 mb-4"
    >
      Solid + Label
    </h2>
     
    <div
      class="base-divider base-divider--solid base-divider--horizontal"
      role="separator"
    >
      <div
        class="base-divider__label"
      >
        or
      </div>
    </div>
     
    <h2
      class="type-overline type-overline-small mt-8 mb-4"
    >
      Dashed + Label
    </h2>
     
    <div
      class="base-divider base-divider--dashed base-divider--horizontal"
      role="separator"
    >
      <div
        class="base-divider__label"
      >
        or
      </div>
    </div>
  </div>
   
  <div
    class="w-64"
  >
    <h1
      class="type-overline type-overline-large"
    >
      Vertical
    </h1>
     
    <div
      class="flex items-end"
    >
      <div
        class="flex-grow w-1/4"
      >
        <h2
          class="type-overline type-overline-small mb-4"
        >
          Solid
        </h2>
         
        <div
          class="base-divider base-divider--solid base-divider--vertical h-56"
          role="separator"
        >
          <!---->
        </div>
      </div>
       
      <div
        class="flex-grow w-1/4"
      >
        <h2
          class="type-overline type-overline-small mb-4"
        >
          Dashed
        </h2>
         
        <div
          class="base-divider base-divider--dashed base-divider--vertical h-56"
          role="separator"
        >
          <!---->
        </div>
      </div>
       
      <div
        class="flex-grow w-1/4"
      >
        <h2
          class="type-overline type-overline-small mb-4"
        >
          Solid + Label
        </h2>
         
        <div
          class="base-divider base-divider--solid base-divider--vertical h-56"
          role="separator"
        >
          <div
            class="base-divider__label"
          >
            or
          </div>
        </div>
      </div>
       
      <div
        class="flex-grow w-1/4"
      >
        <h2
          class="type-overline type-overline-small mb-4"
        >
          Dashed + Label
        </h2>
         
        <div
          class="base-divider base-divider--dashed base-divider--vertical h-56"
          role="separator"
        >
          <div
            class="base-divider__label"
          >
            or
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
